internal/controllers/core/liveupdate/reconciler.go

```diff
@@ -925,63 +925,87 @@ func (r *Reconciler) applyInternal(
+				build.PathMappingsToContainerPaths(toRemove), boiledSteps, hotReload)
+			_ = archive.Close()
+
+			lastFileTimeSynced := input.LastFileTimeSynced
+			if lastFileTimeSynced.IsZero() {
+				lastFileTimeSynced = apis.NowMicro()
+			}
 
-		cStatus := v1alpha1.LiveUpdateContainerStatus{
-			ContainerName:      cInfo.ContainerName.String(),
-			ContainerID:        cInfo.ContainerID.String(),
-			PodName:            cInfo.PodID.String(),
-			Namespace:          cInfo.Namespace.String(),
-			LastFileTimeSynced: lastFileTimeSynced,
-		}
+			cStatus := v1alpha1.LiveUpdateContainerStatus{
+				ContainerName:      cInfo.ContainerName.String(),
+				ContainerID:        cInfo.ContainerID.String(),
+				PodName:            cInfo.PodID.String(),
+				Namespace:          cInfo.Namespace.String(),
+				LastFileTimeSynced: lastFileTimeSynced,
+			}
 
-		if err != nil {
-			if runFail, ok := build.MaybeRunStepFailure(err); ok {
-				// Keep running updates -- we want all containers to have the same files on them
-				// even if the Runs don't succeed
-				logger.Get(ctx).Infof("  → Failed to update container %s: run step %q failed with exit code: %d",
-					cInfo.DisplayName(), runFail.Cmd.String(), runFail.ExitCode)
-				cStatus.LastExecError = err.Error()
-				lastExecErrorStatus = &cStatus
-			} else {
-				// Something went wrong with this update and it's NOT the user's fault--
-				// likely a infrastructure error. Bail, and fall back to full build.
-				result.Failed = &v1alpha1.LiveUpdateStateFailed{
-					Reason:  "UpdateFailed",
-					Message: fmt.Sprintf("Updating pod %s: %v", cStatus.PodName, err),
+			if err != nil {
+				if runFail, ok := build.MaybeRunStepFailure(err); ok {
+					// Keep running updates -- we want all containers to have the same files on them
+					// even if the Runs don't succeed
+					logger.Get(copierContext).Infof("  → Failed to update container %s: run step %q failed with exit code: %d",
+						cInfo.DisplayName(), runFail.Cmd.String(), runFail.ExitCode)
+					cStatus.LastExecError = err.Error()
+
+				} else {
+					// Something went wrong with this update, and it's NOT the user's fault--
+					// likely an infrastructure error. Bail, and fall back to full build.
+					failed = &v1alpha1.LiveUpdateStateFailed{
+						Reason:  "UpdateFailed",
+						Message: fmt.Sprintf("Updating pod %s: %v", cStatus.PodName, err),
+					}
 				}
-				return result
+			} else {
+				logger.Get(copierContext).Infof("  → Container %s updated!", cInfo.DisplayName())
 			}
-		} else {
-			logger.Get(ctx).Infof("  → Container %s updated!", cInfo.DisplayName())
-			if lastExecErrorStatus != nil {
-				// This build succeeded, but previously at least one failed due to user error.
-				// We may have inconsistent state--bail, and fall back to full build.
-				result.Failed = &v1alpha1.LiveUpdateStateFailed{
-					Reason: "PodsInconsistent",
-					Message: fmt.Sprintf("Pods in inconsistent state. Success: pod %s. Failure: pod %s. Error: %v",
```

Comment:
i think you lost this bit

Reply:
So, I'm not sure I fully understand the purpose of this or how to implement in the parallelized version. In the serialized version it would be invoked like:

```
update -> update failure
update -> abort
```

In the new version that logic can't work, since we are doing everything in parallel. However, we do ofc catch errors in the new `select` statement, so we could update the `Reason` field there?...

Reply:
sure!

think about it like this:

- if all `run('yarn install')`s succeed, that's OK
- if all `run('yarn install`)`s fail, that's also OK - it just means you introduced a syntax error
- if some of them succeeded and some of them failed, that's not something we can easily recover from.

an important part of this is there's a distinction between `run("yarn install")` reported some errors in the code being installed, vs `run("yarn install")` failed because of some underlying kubernetes failure.